### PR TITLE
[TextFields] Making a disabled color property on controllers

### PR DESCRIPTION
**Pull request by willlarche**

### CODE REVIEW

**Review by ajsecord — APPROVED** [review]

File: components/TextFields/src/MDCTextInputControllerFullWidth.m [hunk 1]
```diff
@@ -424,6 +424,22 @@ - (void)setCharacterCountViewMode:(UITextFieldViewMode)characterCountViewMode {
   }
 }
 
+- (void)setDisabledColor:(UIColor *)disabledColor {
```
Comment (romoore) [inline components/TextFields/src/MDCTextInputControllerFullWidth.m:427]: mark the parameter as __unused?

File: components/TextFields/src/MDCTextInputControllerFullWidth.m [hunk 2]
```diff
@@ -424,6 +424,22 @@ - (void)setCharacterCountViewMode:(UITextFieldViewMode)characterCountViewMode {
...
+  [self updateUnderline];
+}
+
+- (UIColor *)disabledColor {
+  return [UIColor clearColor];
+}
+
++ (void)setDisabledColorDefault:(UIColor *)disabledColorDefault {
```
Comment (romoore) [inline components/TextFields/src/MDCTextInputControllerFullWidth.m:435]: mark the parameter as __unused?

File: components/TextFields/src/MDCTextInputControllerFullWidth.m (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m:427]: My understanding is you need __unused for suppressing 'unused variable' warnings. This doesn't trigger one in my Xcode 9. Are you seeing it? It's defined in a protocol so it has to be implemented. This particular class doesn't _yet_ need it.

Comment (romoore) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m:427]: I know that __unused can be used to suppress compiler warnings (not sure why it wouldn't be in this case), but I think it's also a healthy call-out to creators and maintainers that there is an argument being discarded.  Unused parameters are often completely innocuous, I just think it's a nice-to-have for source code bookkeeping.

Comment (romoore) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m:427]: I added some unused warnings to the project and I did get a pile of unused parameter complaint from the compiler.

```
-Wunused-comparison -Wunused-const-variable -Wunused-exception-parameter -Wunused-function -Wunused-label -Wunused-member-function -Wunused-parameter -Wunused-private-field -Wunused-property-ivar -Wunused-result -Wunused-value -Wunused-variable -Wunused-volatile-lvalue -Wused-but-marked-unused
```

Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerFullWidth.m:427]: Please propose in weekly meeting. If it's accepted as a team best practice, then let's do it.

**Pull request merged**